### PR TITLE
Add specs and validates

### PR DESCRIPTION
**Pull request by FuadNabi**
PullRequestGIF

# In This Pull Request:🎉💫🚀
### For the User model:

- The name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.

### For the Post model:

- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.

### Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    it 'CommentsCounter must be an integer greater than or equal to zero.' do
+      subject.comments_counter = -1
+      expect(subject).to_not be_valid
+    end
+
+    it 'LikesCounter must be an integer greater than or equal to zero.' do
+      subject.likes_counter = -1
+      expect(subject).to_not be_valid
+    end
+  end
```
Comment (julie-ify) [inline spec/models/post_spec.rb]: - [ ] All methods in the models should be tested. Kindly add tests for the `count_updater` and `recent_comments` methods in your Post model

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  end
+
+  context 'Testing behavior' do
+    user = User.new(name: 'John')
+    user.id = 1
+    user.save
+
+    before { 5.times { Post.create(author_id: user.id, title: 'Hello World', text: 'Text') } }
+    it 'lists the most 3 recent posts' do
+      expect(user.recent_posts.length).to eq 3
```
Comment (julie-ify) [inline spec/models/user_spec.rb]: - [ ] This test appears to be failing. Kindly fix it.

  image

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @FuadNabi,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @FuadNabi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- No linter errors ✔️ 
- Descriptive PR ✔️ 


 ## Required Changes ♻️
_Check the comments under the review._

- [ ] There is an error in the [validations-specs](https://github.com/FuadNabi/blog-app/tree/validations-specs) branch that doesn't let me continue reviewing your required changes, please fix it before submitting again. I proved with the dev branch and it run well. 👇🏽 

image

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 🟢 🟢 🟢

Hi @FuadNabi,

Good job working on the changes asked by the previous reviewers 👍🏽

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights:

✅ You used Gitflow correctly 👍🏽
✅ Linter checks passed 👍🏽
✅ Implement validations for models and tests 👍🏽
✅ Tests for model methods passed  👍🏽

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**